Orange/data/table.py

```diff
@@ -114,24 +114,34 @@ def __setitem__(self, key, value):
     def _str(self, limit):
         def sp_values(matrix, variables):
             if not sp.issparse(matrix):
+                if matrix.ndim == 1:
+                    matrix = matrix[:, np.newaxis]
                 return Instance.str_values(matrix[row], variables, limit)
-            begptr, endptr = matrix.indptr[row:row + 2]
-            rendptr = endptr if not limit else min(endptr, begptr + 5)
-            variables = [variables[var]
-                         for var in matrix.indices[begptr:rendptr]]
-            s = ", ".join(
-                "{}={}".format(var.name, var.str_val(val))
-                for var, val in zip(variables, matrix.data[begptr:rendptr]))
-            if limit and rendptr != endptr:
+
+            # Sparse matrices can't handle slices where an index is out of
+            # bounds like numpy arrays can, so we must determine largest index
+            _, columns = matrix.shape
+
+            row_entries, idx = [], 0
+            while len(row_entries) < 5 and idx < len(variables):
```

Comment:
This doesn't consider the `limit` argument. I.e. only 5 values are printed regardless whether the limit is set to either true or false.

<img width="1015" alt="screen shot 2017-07-15 at 17 19 19" src="https://user-images.githubusercontent.com/713026/28240429-0a0b049e-6982-11e7-85c5-f632eae3d045.png">